### PR TITLE
Export genesis block and open keys

### PR DESCRIPTION
**Pull request by jingsting**
## Purpose
Adding an RPC method for exporting the genesis block and open keys as encoded Base58 strings to make spinning up predefined test networks easier

## Approach
Added a new debug RPC method `debug_exportGenesisAndKeys`

## Tickets
* #2006

### CODE REVIEW

File: common/src/main/scala/co/topl/codecs/binary/scodecs/genesisAndKeys/GenesisAndKeysCodecs.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+trait GenesisAndKeysCodecs {
+
+  implicit val genesisAndKeysCodecs: Codec[GenesisAndKeys] =
+    (blockCodec :: seqCodec(privateKeyCurve25519Codec)).xmapc[GenesisAndKeys] { case genesis :: keys :: HNil =>
+      GenesisAndKeys(genesis, keys)
+    }(genesisAndKeys => HList(genesisAndKeys.genesis, genesisAndKeys.keys))
+}
+
+case class GenesisAndKeys(genesis: Block, keys: Seq[PrivateKeyCurve25519])
```
Comment (b-vennes) [inline common/src/main/scala/co/topl/codecs/binary/scodecs/genesisAndKeys/GenesisAndKeysCodecs.scala:19]: We might want to find a different place to put this. Somewhere closer to the RPC

File: common/src/main/scala/co/topl/codecs/binary/scodecs/genesisAndKeys/GenesisAndKeysCodecs.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import co.topl.codecs.binary.scodecs.modifier.blockCodec
+import co.topl.codecs.binary.scodecs.valuetypes._
+import co.topl.modifier.block.Block
+import scodec.Codec
+import shapeless.{::, HList, HNil}
+
+trait GenesisAndKeysCodecs {
+
+  implicit val genesisAndKeysCodecs: Codec[GenesisAndKeys] =
+    (blockCodec :: seqCodec(privateKeyCurve25519Codec)).xmapc[GenesisAndKeys] { case genesis :: keys :: HNil =>
```
Comment (b-vennes) [inline common/src/main/scala/co/topl/codecs/binary/scodecs/genesisAndKeys/GenesisAndKeysCodecs.scala:14]: I think we could change this to `(blockCodec :: seqCodec(privateKeyCurve25519Codec)).as[GenesisAndKeys]`

File: common/src/main/scala/co/topl/codecs/json/genesisAndKeys/GenesisAndKeysJsonCodecs.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import io.circe.{Decoder, Encoder}
+import scodec.Codec
+import scodec.bits.BitVector
+
+trait GenesisAndKeysJsonCodecs {
+
+  implicit val genesisAndKeysJsonEncoder: Encoder[GenesisAndKeys] = { genesisAndKeys =>
+    Codec[GenesisAndKeys]
+      .encode(genesisAndKeys)
+      .getOrElse(throw new Exception("Unable to encode genesis block and keys"))
```
Comment (b-vennes) [inline common/src/main/scala/co/topl/codecs/json/genesisAndKeys/GenesisAndKeysJsonCodecs.scala:19]: I think we can handle the exception inside the context of the `Encoder`. I believe it returns some sort of Result type

File: node/src/main/scala/co/topl/rpc/handlers/DebugRpcHandlerImpls.scala [hunk 4]
```diff
@@ -61,6 +63,22 @@ class DebugRpcHandlerImpls(
...
         )
 
+  override val exportGenesisAndKeys: ToplRpc.Debug.ExportGenesisAndKeys.rpc.ServerHandler =
+    _ =>
+      for {
+        block <- withNodeView(_.history.modifierByHeight(1)).subflatMap(_.toRight(ToplRpcErrors.NoBlockAtHeight))
+        secrets <- keyManagerInterface.getOpenKeys
+          .leftMap(e => ToplRpcErrors.genericFailure(e.toString): RpcError)
+      } yield {
+        val secretList = secrets.flatMap {
```
Comment (b-vennes) [inline node/src/main/scala/co/topl/rpc/handlers/DebugRpcHandlerImpls.scala:73]: I think we could do this via a `.collect` on the result of `getOpenKeys`

File: topl-rpc/src/main/scala/co/topl/rpc/ToplRpc.scala [hunk 5]
```diff
@@ -66,6 +67,16 @@ object ToplRpc {
...
     }
+
+    object ExportGenesisAndKeys {
+
+      /**
+       * Export the genesis block and open keys as Json files on disk
+       */
+      val rpc: Rpc[Params, Response] = Rpc("debug_exportGenesisAndKeys")
+      case class Params()
+      case class Response(genesisAndKeys: GenesisAndKeys)
```
Comment (b-vennes) [inline topl-rpc/src/main/scala/co/topl/rpc/ToplRpc.scala:78]: Do we want this response type to be `Response(genesis: Block, keys: Set[_ <: Secret])`?

Comment (SeanCheatham) [reply topl-rpc/src/main/scala/co/topl/rpc/ToplRpc.scala:78]: I spoke with James about this.  The intention of this RPC is not necessarily to surface the block and keys separately.  We instead want to return a binary-encoded string containing those two items together.

**Review by b-vennes — APPROVED** [review]

**Review by scasplte2 — APPROVED** [review]
👍

**Pull request merged**